src/fwrite.c

```diff
@@ -829,9 +829,28 @@ void fwriteMain(fwriteMainArgs args)
            zbuffSize / MEGA, nth, errno, strerror(errno));
       // # nocov end
     }
-  }
+    len = 0;
+    crc = crc32(0L, Z_NULL, 0);
 
-#endif
+    if (f != -1) {
+      // write minimal gzip header, but not on the console
+      static const char header[] = "\037\213\10\0\0\0\0\0\0\3";
+      int ret0 = WRITE(f, header, (sizeof header) - 1);
+      compress_len += (sizeof header) - 1;
+
+      if (ret0 == -1) {
+        // # nocov start
+        int errwrite = errno; // capture write errno now in case close fails with a different errno
```

Comment:
we can remove this right?

```suggestion
```

Reply:
Good catch! Made use of the `errno` instead of just saying that `write()` returned -1. Alternatively we may change the translation template and make use of `strerror()` as below:
https://github.com/Rdatatable/data.table/blob/551aa8737f9dc4a7780c707de443422467a7322b/src/fwrite.c#L924

Reply:
LGTM for now